src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -294,11 +294,10 @@ public TagHelperContent EndTagHelperWritingScope()
 
                 foreach (var attribute in tagHelperOutput.Attributes)
                 {
-                    var value = HtmlEncoder.HtmlEncode(attribute.Value);
                     writer.Write(' ');
                     writer.Write(attribute.Key);
                     writer.Write("=\"");
-                    writer.Write(value);
+                    WriteTo(writer, HtmlEncoder, attribute.Value, escapeQuotes: true);
```

Comment:
- Hmm? I think you misunderstood what I meant. Don't think that should exist inside of `HtmlString`, instead of doing `WriteTo(writer, HtmlEncoder, attribute.Value, escapeQuotes: true);` you'd do:
```C#
var attributeValue = attribute.Value as HtmlString;
if (attributeValue != null && attributeValue.Value.Contains("\"", StringComparison.Ordinal))
{
    attributeValue = new HtmlString(attributeValue.Value.Replace("\"", "&quote;"));
    WriteTo(writer, attributeValue);
}
else
{
    WriteTo(writer, attribute.Value);
}
```
- :ok: